File: backend/src/BloodDonationNetwork.Domain/Entities/DonorProfile.cs
namespace BloodDonationNetwork.Domain.Entities;

public class DonorProfile
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid UserId { get; set; }
    public string BloodType { get; set; } = default!;
    public string EligibilityStatus { get; set; } = "pending_review";
    public DateOnly DateOfBirth { get; set; }
    public DateOnly? LastDonationDate { get; set; }
    public string? Address { get; set; }
    public double? Latitude { get; set; }
    public double? Longitude { get; set; }
    public bool LocationVerified { get; set; } = false;
    public Dictionary<string, bool> MedicalFlags { get; set; } = new();
    public bool VerifiedByAdmin { get; set; } = false;
    public decimal ReliabilityScore { get; set; } = 1.0m;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: mobile/lib/features/donor_profile/models/donor_profile.dart
class DonorProfile {
  const DonorProfile({
    required this.id,
    required this.userId,
    required this.fullName,
    required this.bloodType,
    required this.eligibilityStatus,
    required this.dateOfBirth,
    this.lastDonationDate,
    this.address,
    this.medicalFlags = const {},
    this.latitude,
    this.longitude,
    required this.locationVerified,
    required this.verifiedByAdmin,
  });

  final String id;
  final String userId;
  final String fullName;
  final String bloodType;
  final String eligibilityStatus;
  final DateTime dateOfBirth;
  final DateTime? lastDonationDate;
  final String? address;
  final Map<String, bool> medicalFlags;
  final double? latitude;
  final double? longitude;
  final bool locationVerified;
  final bool verifiedByAdmin;

  factory DonorProfile.fromJson(Map<String, dynamic> json) => DonorProfile(
        id: json['id'] as String,
        userId: json['userId'] as String,
        fullName: json['fullName'] as String? ?? '',
        bloodType: json['bloodType'] as String,
        eligibilityStatus: json['eligibilityStatus'] as String,
        dateOfBirth: DateTime.parse(json['dateOfBirth'] as String),
        lastDonationDate: json['lastDonationDate'] == null
            ? null
            : DateTime.parse(json['lastDonationDate'] as String),
        address: json['address'] as String?,
        medicalFlags: Map<String, bool>.from(json['medicalFlags'] as Map? ?? {}),
        latitude: (json['latitude'] as num?)?.toDouble(),
        longitude: (json['longitude'] as num?)?.toDouble(),
        locationVerified: json['locationVerified'] as bool? ?? false,
        verifiedByAdmin: json['verifiedByAdmin'] as bool? ?? false,
      );
}